Source/Scene/Cesium3DTile.js

```diff
@@ -0,0 +1,1018 @@
+
+    /**
+     * Assuming the tile's bounding volume intersects the culling volume, determines
+     * whether the tile's content's bounding volume intersects the culling volume.
+     *
+     * @param {FrameState} frameState The frame state.
+     * @returns {Intersect} The result of the intersection: the tile's content is completely outside, completely inside, or intersecting the culling volume.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.contentVisibility = function(frameState) {
+        // Assumes the tile's bounding volume intersects the culling volume already, so
+        // just return Intersect.INSIDE if there is no content bounding volume.
+        if (!defined(this._contentBoundingVolume)) {
+            return Intersect.INSIDE;
+        }
+
+        // PERFORMANCE_IDEA: is it possible to burn less CPU on this test since we know the
+        // tile's (not the content's) bounding volume intersects the culling volume?
+        var cullingVolume = frameState.cullingVolume;
+        var boundingVolume = getContentBoundingVolume(this, frameState);
+        return cullingVolume.computeVisibility(boundingVolume);
+    };
+
+    /**
+     * Computes the (potentially approximate) distance from the closest point of the tile's bounding volume to the camera.
+     *
+     * @param {FrameState} frameState The frame state.
+     * @returns {Number} The distance, in meters, or zero if the camera is inside the bounding volume.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.distanceToTile = function(frameState) {
+        var boundingVolume = getBoundingVolume(this, frameState);
+        return boundingVolume.distanceToCamera(frameState);
+    };
+
+    var scratchCartesian = new Cartesian3();
+
+    /**
+     * Computes the distance from the center of the tile's bounding volume to the camera.
+     *
+     * @param {FrameState} frameState The frame state.
+     * @returns {Number} The distance, in meters, or zero if the camera is inside the bounding volume.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.distanceToTileCenter = function(frameState) {
+        var boundingVolume = getBoundingVolume(this, frameState).boundingVolume;
+        var toCenter = Cartesian3.subtract(boundingVolume.center, frameState.camera.positionWC, scratchCartesian);
+        var distance = Cartesian3.magnitude(toCenter);
+        Cartesian3.divideByScalar(toCenter, distance, toCenter);
+        var dot = Cartesian3.dot(frameState.camera.directionWC, toCenter);
+        return distance * dot;
+    };
+
+    /**
+     * Checks if the camera is inside the viewer request volume.
+     *
+     * @param {FrameState} frameState The frame state.
+     * @returns {Boolean} Whether the camera is inside the volume.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.insideViewerRequestVolume = function(frameState) {
+        var viewerRequestVolume = this._viewerRequestVolume;
+        if (!defined(viewerRequestVolume)) {
```

Comment:
You can shorten this function to 

`!defined(viewerRequestVolume) || (viewerRequestVolume.distanceToCamera(frameState) === 0.0)`